janus_server/src/aggregator.rs

```diff
@@ -1751,42 +1758,59 @@ fn aggregator_filter<C: Clock>(
         .init();
 
     let hpke_config_endpoint = warp::path("hpke_config")
-        .and(warp::get())
-        .and(with_cloned_value(aggregator.clone()))
-        .and(warp::query::<HashMap<String, String>>())
-        .then(
-            |aggregator: Arc<Aggregator<C>>, query_params: HashMap<String, String>| async move {
-                let task_id_b64 = query_params
-                    .get("task_id")
-                    .ok_or(Error::UnrecognizedMessage("task_id", None))?;
-                let hpke_config_bytes = aggregator.handle_hpke_config(task_id_b64.as_ref()).await?;
-                http::Response::builder()
-                    .header(CACHE_CONTROL, "max-age=86400")
-                    .header(CONTENT_TYPE, HpkeConfig::MEDIA_TYPE)
-                    .body(hpke_config_bytes)
-                    .map_err(|err| Error::Internal(format!("couldn't produce response: {}", err)))
-            },
+        .and(
+            warp::get()
+                .and(with_cloned_value(aggregator.clone()))
+                .and(warp::query::<HashMap<String, String>>())
+                .then(
+                    |aggregator: Arc<Aggregator<C>>, query_params: HashMap<String, String>| async move {
+                        let task_id_b64 = query_params
+                            .get("task_id")
+                            .ok_or(Error::UnrecognizedMessage("task_id", None))?;
+                        let hpke_config_bytes = aggregator.handle_hpke_config(task_id_b64.as_ref()).await?;
+                        http::Response::builder()
+                            .header(CACHE_CONTROL, "max-age=86400")
+                            .header(CONTENT_TYPE, HpkeConfig::MEDIA_TYPE)
+                            .body(hpke_config_bytes)
+                            .map_err(|err| Error::Internal(format!("couldn't produce response: {}", err)))
+                    },
+                )
+                .map(error_handler(&response_counter, "hpke_config"))
+                .with(
+                    warp::cors()
+                        .allow_any_origin()
+                        .allow_method("GET")
+                        .max_age(CORS_PREFLIGHT_CACHE_AGE),
+                ),
```

Comment:
nit: I wonder if common stuff like these, setting up the timing wrapper and the named `tracing` spans could be factored out into a common function so that we don't have to repeat it across every HTTP endpoint.

Reply:
Yeah, that could be a good improvement. Maybe we could have a utility function take a routing filter, a response-making filter, and a name/label, and return a composite filter with appropriate wrappers. I'll try this out as part of #211.